src/mlpack/core/util/arma_traits.hpp

```diff
@@ -49,6 +49,19 @@ struct IsCube
   static const bool value = false;
 };
 
+template<typename FieldType>
+struct IsField
+{
+  static const bool value = false;
+};
+
+template<typename T>
+struct IsAnyArmaBaseType
+{
+  static const bool value = IsVector<T>::value || IsMatrix<T>::value ||
+      IsCube<T>::value || IsField<T>::value;
+};
```

Comment:
Just a note, these were needed because `arma::is_arma_type` is actually not true for `arma::field`.